udf-compiler/src/test/scala/com/nvidia/spark/OpcodeSuite.scala

```diff
@@ -45,19 +42,1792 @@ class OpcodeSuite extends FunSuite {
+    val result = dataset.withColumn("new", u(col("x"),col("y")))
+    val ref = dataset.withColumn("new", lit(col("x").equalTo(col("y"))))
+    checkEquiv(result, ref)
+  }
+
+  test("string test - length") {
+    val myudf: (String) => Int = a => {
+      a.length()
+    }
+    val u = makeUdf(myudf)
+    val dataset = List("rocket","").toDF("x").repartition(1)
+    val result = dataset.withColumn("new", u(col("x")))
+    val ref = dataset.withColumn("new", length(col("x")))
+    checkEquiv(result, ref)
+  }
+
+  test("string test - isEmpty") {
+    val myudf: (String) => Boolean = a => {
+      a.isEmpty()
+    }
+    val u = makeUdf(myudf)
+    val dataset = List("rocket","").toDF("x").repartition(1)
+    val result = dataset.withColumn("new", u(col("x")))
+    val ref = dataset.withColumn("new", lit(length(col("x")).equalTo(0)))
+    checkEquiv(result, ref)
+  }
+
+  test("string test - valueOf - Boolean") {
+    val myudf: (Boolean) => String = a => {
+      String.valueOf(a)
+    }
+    val u = makeUdf(myudf)
+    val dataset = List(true).toDF("x").repartition(1)
+    val result = dataset.withColumn("new", u(col("x")))
+    val ref = dataset.withColumn("new", lit("true"))
+    checkEquiv(result, ref)
+  }
+
+  test("string test - valueOf - Char") {
+    val myudf: (Char) => String = a => {
+      String.valueOf(a)
+    }
+    val u = makeUdf(myudf)
+    val dataset = List(5).toDF("x").repartition(1)
+    val result = dataset.withColumn("new", u(col("x")))
+    val ref = dataset.withColumn("new", lit("5"))
+    checkEquiv(result, ref)
+  }
+
+  test("string test - valueOf - Double") {
+    val myudf: (Double) => String = a => {
+      String.valueOf(a)
+    }
+    val u = makeUdf(myudf)
+    val dataset = List(2.0).toDF("x").repartition(1)
+    val result = dataset.withColumn("new", u(col("x")))
+    val ref = dataset.withColumn("new", lit("2.0"))
+    checkEquiv(result, ref)
+  }
+
+  test("string test - valueOf - Float") {
+    val myudf: (Float) => String = a => {
+      String.valueOf(a)
+    }
+    val u = makeUdf(myudf)
+    val dataset = List(2.0f).toDF("x").repartition(1)
+    val result = dataset.withColumn("new", u(col("x")))
+    val ref = dataset.withColumn("new", lit("2.0"))
+    checkEquiv(result, ref)
+  }
+
+  test("string test - valueOf - Int") {
+    val myudf: (Int) => String = a => {
+      String.valueOf(a)
+    }
+    val u = makeUdf(myudf)
+    val dataset = List(2).toDF("x").repartition(1)
+    val result = dataset.withColumn("new", u(col("x")))
+    val ref = dataset.withColumn("new", lit("2"))
+    checkEquiv(result, ref)
+  }
+
+  test("string test - valueOf - Long") {
+    val myudf: (Long) => String = a => {
+      String.valueOf(a)
+    }
+    val u = makeUdf(myudf)
+    val dataset = List(2L).toDF("x").repartition(1)
+    val result = dataset.withColumn("new", u(col("x")))
+    val ref = dataset.withColumn("new", lit("2"))
+    checkEquiv(result, ref)
+  }
+
+  test("string test - charAt") {
```

Comment:
nit: could we make the tests that have `checkEquivNotCompiled` have a standard name like "FALLBACK TO CPU: string test -charAt" in this case.